xcube_stac/store.py

```diff
@@ -134,35 +213,74 @@ def get_data_ids(
                 is returned. So far only the attribute 'title' is supported.
                 Defaults to None.
 
-        Returns:
+        Yields:
             An iterator over the identifiers (and additional attributes defined
             by *include_attrs* of data resources provided by this data store).
         """
-        self._assert_valid_data_type(data_type)
-        return self.stac.get_data_ids(
-            items=items,
-            item_data_ids=item_data_ids,
-            include_attrs=include_attrs,
-            **open_params
-        )
+        if data_type is not None:
+            warnings.warn(
+                f'data_type is set to {data_type}, but is not used.'
+            )
+        if items is None:
+            items, item_data_ids = self.get_item_collection(**open_params)
+        if item_data_ids is None:
+            item_data_ids = self.get_item_data_ids(items)
+
+        for (item, item_data_id) in zip(items, item_data_ids):
+            for asset in self.get_assets_from_item(
+                item, include_attrs, **open_params
+            ):
+                if include_attrs is not None:
+                    (asset, attrs) = asset
+                    data_id = (
+                        item_data_id + self._data_id_delimiter + asset
+                    )
+                    yield (data_id, attrs)
+                else:
+                    data_id = item_data_id + self._data_id_delimiter + asset
+                    yield data_id
 
     def has_data(self, data_id: str, data_type: DataTypeLike = None) -> bool:
         if self._is_valid_data_type(data_type):
             return data_id in self.list_data_ids()
         return False
 
-    def describe_data(self, data_id: str, **open_params) -> DataDescriptor:
-        """Get the descriptor for the data resource given by *data_id*.
+    def get_assets_from_item(
```

Comment:
Is this part of the `SmosDataStore`'s API? It should not as it uncovers an implementation detail, namely that we pystac. Therefore you should make it private by prefixing with `_`.

Reply:
I made it private.